### DISCUSSION TITLE
SVG option?

### DISCUSSION THREAD

**Discussion by bryanjamesmiller** [stefanzweifel/sidecar-browsershot#22] [Ideas]
Thanks for the awesome package.  Is there possibly a way you could add an option to save the output as an SVG?  That would be super awesome!!

**Comment by stefanzweifel** [comment]
Hey

I'm not sure how this would work. What would you pass to Browsershot? An URL? HTML and CSS?
Would the API look something like this?

```php
$svg = BrowsershotLambda::html('<div>Hello World</div>')->bodySvg();
        
Storage::disk('public')->put('example.svg', $svg);
```

I wanted to point out that this isn't something that should be solved in sidecar-browsershot, but in Browsershot itself. But I saw you already opened [a discussion](https://github.com/spatie/browsershot/discussions/655) and didn't get an answer yet.

But Browsershot itself is also just a wrapper around Puppeteer, and Puppeteer doesn't have that feature … as far as I can see from the [documentation](https://pptr.dev/).
There is also no issue or discussion for this: https://github.com/puppeteer/puppeteer/search?p=1&q=svg&type=issues

Not sure how to go forward with this. I surely don't have the knowledge on how to transform HTML to SVG.

**Reply by bryanjamesmiller** [reply]
Thanks for the response @stefanzweifel - and for looking into this and explaining it to me.  The business requirement has since gone away as we are now allowed to use PNG, so it is no longer needed for us - although we could still use it if Spatie decides to implement it somehow.  Your package is a dream - thanks again!!
